exchanges/kraken.py

```diff
@@ -19,16 +20,46 @@ def latest_trade(self, asset_pair):
 
         ticker = self._public_query('Ticker', {'pair': [asset_pair]})
         price, volume = ticker[asset_pair]['c']
-        return Trade(asset_pair, price, volume)
+        return Trade(asset_pair, price, volume, time.time())
 
     def place_order(self, order):
         pass
 
     def order_info(self, order_id):
         pass
 
-    def trades_history(self, asset_pair, start_time, finish_time):
-        pass
+    def trades_history(self, asset_pair, since, until=None):
+        """Retrieves all the trades of an asset pair made between
+           two certain dates"""
+        # We can only retrieve the first 1000 trades made after a
+        # given date, so maybe we need to make multiple queries
+        # to obtain the desired period of time. Please note that
+        # the precision used in the 'since' parameter of the query
+        # is higher than the one used by unix
+        response = self._public_query('Trades', {'pair':asset_pair,
+                                                 'since':since*10**9})
+        trades = response[asset_pair]
+        next_id = int(response['last'])
+
+        if until is None:
+            until = until_kraken = next_id
+        else:
+            until_kraken = int(until*10**9)  # Kraken's timestamp precision
+ 
+        while next_id < until_kraken:
+            # The second time we make a query, we can use the
+            # 'last ID' of the previous query as the parameter 'since'.
+            # This is recommended in the api's documentation
+            response = self._public_query('Trades', {'pair':asset_pair,
```

Comment:
For what I understand, only private queries have a rate limit. Anyway, with the current implementation we cannot hit the limit. Each call to response is really slow (since the connection is not persistent).